#### 1. Import pandas library

In [15]:
import pandas as pd

#### 2. Import mysql.connector


In [2]:
import mysql.connector

#### 2.2. Run the following line


In [20]:
cnx = mysql.connector.connect(user='guest', password='relational', host='relational.fit.cvut.cz', database='stats', use_pure=True)

### Import users table:

#### 3. Initialize cursor (as seen in classes' Jupyter Notebook)

In [21]:
cursor = cnx.cursor()

#### 4. Initialize query  (as seen in classes' Jupyter Notebook) 
#### You want the users table from stats database. (Hint: stats.users)

In [22]:
query = ("SELECT * FROM users")  ## Any query

#### 5. Send query to database  (as seen in classes' Jupyter Notebook)

In [23]:
cursor.execute(query)

#### 6. Create a new dataframe (preferably called users) with the following columns names (as seen in classes' Jupyter Notebook)

In [24]:
columns_users = ['Id', 'Reputation', 'CreationDate', 'DisplayName', 'LastAccessDate', 'WebsiteUrl', 'Location', 'AboutMe', 'Views', 'UpVotes', 'DownVotes', 'AccountId', 'Age', 'ProfileImageUrl']
users_ds = pd.DataFrame(columns = columns_users)

users_ds.head(10)

Empty DataFrame
Columns: [Id, Reputation, CreationDate, DisplayName, LastAccessDate, WebsiteUrl, Location, AboutMe, Views, UpVotes, DownVotes, AccountId, Age, ProfileImageUrl]
Index: []

#### 7. Fill the new users dataframe with the data you can get iterating through cursor (as seen in classes' Jupyter Notebook)

In [25]:
for i, data in enumerate(cursor):
    users_ds.loc[i] = data

users_ds.head(23)

Id Reputation        CreationDate         DisplayName      LastAccessDate  \
0   -1          1 2010-07-19 06:55:26           Community 2010-07-19 06:55:26   
1    2        101 2010-07-19 14:01:36        Geoff Dalgas 2013-11-12 22:07:23   
2    3        101 2010-07-19 15:34:50        Jarrod Dixon 2014-08-08 06:42:58   
3    4        101 2010-07-19 19:03:27              Emmett 2014-01-02 09:31:02   
4    5       6792 2010-07-19 19:03:57               Shane 2014-08-13 00:23:47   
5    6        457 2010-07-19 19:04:07              Harlan 2014-08-07 19:49:44   
6    7        429 2010-07-19 19:04:37               Vince 2014-09-10 21:14:12   
7    8       6764 2010-07-19 19:04:52         csgillespie 2014-09-09 21:15:08   
8   10        121 2010-07-19 19:05:40              Pierre 2014-07-28 18:15:02   
9   11        136 2010-07-19 19:06:02            wahalulu 2014-07-23 21:28:18   
10  12        101 2010-07-19 19:06:34                 Jin 2014-09-09 20:26:03   
11  13        817 2010-07-19 19:06:49             Sharpie 2014-08-14 17:28:14   
12  15         11 2010-07-19 19:07:32       hannes.koller 2010-08-24 14:23:18   
13  16        101 2010-07-19 19:08:13           slashnick 2010-08-19 20:40:59   
14  17        345 2010-07-19 19:08:28              Random 2010-09-10 07:34:36   
15  18        128 2010-07-19 19:08:29              grokus 2012-08-09 15:02:00   
16  19        101 2010-07-19 19:08:45         Noah Snyder 2012-06-17 15:53:43   
17  21        101 2010-07-19 19:08:56  Waleed Al-Balooshi 2011-09-26 02:28:45   
18  22        591 2010-07-19 19:09:00               radek 2014-09-11 15:21:01   
19  23        308 2010-07-19 19:09:16         Jay Stevens 2014-08-11 19:51:32   
20  24        344 2010-07-19 19:09:32              A Lion 2011-12-06 04:21:56   
21  25       4968 2010-07-19 19:09:32     Harvey Motulsky 2014-09-13 21:51:42   
22  26       3150 2010-07-19 19:09:39                 JMS 2014-09-13 04:03:25   

                                 WebsiteUrl                   Location  \
0            http://meta.stackexchange.com/         on the server farm   
1                  http://stackoverflow.com              Corvallis, OR   
2                  http://stackoverflow.com               New York, NY   
3              http://minesweeperonline.com          San Francisco, CA   
4                   http://www.statalgo.com               New York, NY   
5             http://www.harlan.harris.name       District of Columbia   
6         http://bioinformatics.ucdavis.edu                  Davis, CA   
7          http://www.mas.ncl.ac.uk/~ncsg3/  Newcastle, United Kingdom   
8           http://plindenbaum.blogspot.com                     France   
9   http://www.linkedin.com/in/marckvaisman             Washington, DC   
10                      http://www.8164.org                Raleigh, NC   
11                http://www.sharpsteen.net              United States   
12                    http://soma.denkt.org            Vienna, Austria   
13                                     None     London, United Kingdom   
14                                     None                       None   
15                     http://wikipedia.org              United States   
16           http://sbseminar.wordpress.com               New York, NY   
17                                     None                   Michigan   
18                                     None         Berne, Switzerland   
19          http://www.centerlinescores.com                       None   
20                                     None                       None   
21                  http://www.graphpad.com            Santa Monica CA   
22                                     None                       None   

                                              AboutMe Views UpVotes DownVotes  \
0   <p>Hi, I'm not really a person.</p>\n\n<p>I'm ...     0    5007      1920   
1   <p>Developer on the StackOverflow team.  Find ...    25       3         0   
2   <p><a href="http://blog.stackoverfl

#### 8. Rename Id column to userId

In [109]:
users_ds = users_ds.rename(columns={"Id": "userId"})

### Import posts table:

#### 9. Initialize cursor (as seen in classes' Jupyter Notebook)

#### 10. Initialize query  (as seen in classes' Jupyter Notebook) 
#### You want the posts table from stats database. (Hint: no hint this time ;) )

#### 11. Send query to database  (as seen in classes' Jupyter Notebook)

#### 12. Create a new dataframe (preferably called posts) with the following columns names (as seen in classes' Jupyter Notebook)

In [125]:
columns_posts = ['Id', 'PostTypeId', 'AcceptedAnswerId', 'CreaionDate', 'Score', 'ViewCount', 'Body', 'OwnerUserId', 'LasActivityDate', 'Title', 'Tags', 'AnswerCount', 'CommentCount', 'FavoriteCount', 'LastEditorUserId', 'LastEditDate', 'CommunityOwnedDate', 'ParentId', 'ClosedDate', 'OwnerDisplayName', 'LastEditorDisplayName']

In [127]:
posts_ds = pd.DataFrame(columns = columns_posts)

posts_ds.head()

Empty DataFrame
Columns: [Id, PostTypeId, AcceptedAnswerId, CreaionDate, Score, ViewCount, Body, OwnerUserId, LasActivityDate, Title, Tags, AnswerCount, CommentCount, FavoriteCount, LastEditorUserId, LastEditDate, CommunityOwnedDate, ParentId, ClosedDate, OwnerDisplayName, LastEditorDisplayName]
Index: []

[0 rows x 21 columns]

#### 12. Fill the new posts dataframe with the data you can get iterating through cursor (as seen in classes' Jupyter Notebook)

In [128]:
posts_ds = pd.read_csv("/home/inrx/Ironhack/labs/lab-data-cleaning/your-code/posts_table.csv") 

posts_ds.head()

Id  PostTypeId  AcceptedAnswerId          CreaionDate  Score  ViewCount  \
0   1           1              15.0  2010-07-19 19:12:12     23     1278.0   
1   2           1              59.0  2010-07-19 19:12:57     22     8198.0   
2   3           1               5.0  2010-07-19 19:13:28     54     3613.0   
3   4           1             135.0  2010-07-19 19:13:31     13     5224.0   
4   5           2               NaN  2010-07-19 19:14:43     81        NaN   

                                                Body  OwnerUserId  \
0  <p>How should I elicit prior distributions fro...          8.0   
1  <p>In many different statistical methods there...         24.0   
2  <p>What are some valuable Statistical Analysis...         18.0   
3  <p>I have two groups of data.  Each with a dif...         23.0   
4  <p>The R-project</p>\n\n<p><a href="http://www...         23.0   

       LasActivityDate                                              Title  \
0  2010-09-15 21:08:26                      Eliciting priors from experts   
1  2012-11-12 09:21:54                                 What is normality?   
2  2013-05-27 14:48:36  What are some valuable Statistical Analysis op...   
3  2010-09-08 03:00:19  Assessing the significance of differences in d...   
4  2010-07-19 19:21:15                                                NaN   

           ...          AnswerCount  CommentCount  FavoriteCount  \
0          ...                  5.0             1           14.0   
1          ...                  7.0             1            8.0   
2          ...                 19.0             4           36.0   
3          ...                  5.0             2            2.0   
4          ...                  NaN             3            NaN   

   LastEditorUserId         LastEditDate   CommunityOwnedDate ParentId  \
0               NaN                  NaN                  NaN      NaN   
1              88.0  2010-08-07 17:56:44                  NaN      NaN   
2             183.0  2011-02-12 05:50:03  2010-07-19 19:13:28      NaN   
3               NaN                  NaN                  NaN      NaN   
4              23.0  2010-07-19 19:21:15  2010-07-19 19:14:43      3.0   

   ClosedDate OwnerDisplayName LastEditorDisplayName  
0         NaN              NaN                   NaN  
1         NaN              NaN                   NaN  
2         NaN              NaN                   NaN  
3         NaN              NaN                   NaN  
4         NaN              NaN                   NaN  

[5 rows x 21 columns]

#### 13. Rename Id column to postId and OwnerUserId to userId

In [129]:
posts_ds = posts_ds.rename(columns={"Id": "postId", "OwnerUserId": "userId"})

posts_ds.head()

postId  PostTypeId  AcceptedAnswerId          CreaionDate  Score  \
0       1           1              15.0  2010-07-19 19:12:12     23   
1       2           1              59.0  2010-07-19 19:12:57     22   
2       3           1               5.0  2010-07-19 19:13:28     54   
3       4           1             135.0  2010-07-19 19:13:31     13   
4       5           2               NaN  2010-07-19 19:14:43     81   

   ViewCount                                               Body  userId  \
0     1278.0  <p>How should I elicit prior distributions fro...     8.0   
1     8198.0  <p>In many different statistical methods there...    24.0   
2     3613.0  <p>What are some valuable Statistical Analysis...    18.0   
3     5224.0  <p>I have two groups of data.  Each with a dif...    23.0   
4        NaN  <p>The R-project</p>\n\n<p><a href="http://www...    23.0   

       LasActivityDate                                              Title  \
0  2010-09-15 21:08:26                      Eliciting priors from experts   
1  2012-11-12 09:21:54                                 What is normality?   
2  2013-05-27 14:48:36  What are some valuable Statistical Analysis op...   
3  2010-09-08 03:00:19  Assessing the significance of differences in d...   
4  2010-07-19 19:21:15                                                NaN   

           ...          AnswerCount  CommentCount  FavoriteCount  \
0          ...                  5.0             1           14.0   
1          ...                  7.0             1            8.0   
2          ...                 19.0             4           36.0   
3          ...                  5.0             2            2.0   
4          ...                  NaN             3            NaN   

   LastEditorUserId         LastEditDate   CommunityOwnedDate ParentId  \
0               NaN                  NaN                  NaN      NaN   
1              88.0  2010-08-07 17:56:44                  NaN      NaN   
2             183.0  2011-02-12 05:50:03  2010-07-19 19:13:28      NaN   
3               NaN                  NaN                  NaN      NaN   
4              23.0  2010-07-19 19:21:15  2010-07-19 19:14:43      3.0   

   ClosedDate OwnerDisplayName LastEditorDisplayName  
0         NaN              NaN                   NaN  
1         NaN              NaN                   NaN  
2         NaN              NaN                   NaN  
3         NaN              NaN                   NaN  
4         NaN              NaN                   NaN  

[5 rows x 21 columns]

In [132]:
posts_ds.describe()

postId    PostTypeId  AcceptedAnswerId         Score  \
count  40000.000000  40000.000000       7266.000000  40000.000000   
mean   22709.318350      1.636175      23186.498211      4.086500   
std    13763.536209      0.594277      14994.186353      6.589488   
min        1.000000      1.000000          5.000000    -19.000000   
25%    10883.750000      1.000000      10936.500000      1.000000   
50%    22050.500000      2.000000      21794.500000      2.000000   
75%    33536.500000      2.000000      33589.250000      5.000000   
max    48325.000000      7.000000     114363.000000    192.000000   

           ViewCount        userId   AnswerCount  CommentCount  FavoriteCount  \
count   15895.000000  38962.000000  15895.000000  40000.000000    7293.000000   
mean     1206.583202   6079.063087      1.594401      2.011525       3.360346   
std      3872.525910   5224.896435      2.055572      2.673990       7.637988   
min        14.000000     -1.000000      0.000000      0.000000       0.000000   
25%       174.000000   1317.000000      1.000000      0.000000       1.000000   
50%       387.000000   4856.000000      1.000000      1.000000       2.000000   
75%       978.000000   9651.000000      2.000000      3.000000       3.000000   
max    175495.000000  55226.000000    136.000000     45.000000     233.000000   

       LastEditorUserId      ParentId  
count      19725.000000  23569.000000  
mean        4635.762180  19986.577666  
std         5751.397127  13752.497021  
min           -1.000000      1.000000  
25%          702.000000   7869.000000  
50%         2116.000000  18423.000000  
75%         7290.000000  30723.000000  
max        53985.000000  48314.000000

#### 14. Define new dataframes for users and posts with the following selected columns:
    **users columns**: userId, Reputation,Views,UpVotes,DownVotes
    **posts columns**: postId, Score,userID,ViewCount,CommentCount

In [134]:
users_ds = users_ds[["userId", "Reputation","Views","UpVotes","DownVotes"]]
users_ds.head()

userId Reputation Views UpVotes DownVotes
0     -1          1     0    5007      1920
1      2        101    25       3         0
2      3        101    22      19         0
3      4        101    11       0         0
4      5       6792  1145     662         5

In [135]:
posts_ds = posts_ds[["postId", "Score", "userId","ViewCount","CommentCount"]]
posts_ds.head()

postId  Score  userId  ViewCount  CommentCount
0       1     23     8.0     1278.0             1
1       2     22    24.0     8198.0             1
2       3     54    18.0     3613.0             4
3       4     13    23.0     5224.0             2
4       5     81    23.0        NaN             3

#### 15. Merge both dataframes, users and posts. 
You will need to make a [merge](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.merge.html) of posts and users dataframes.

In [136]:
users_posts_ds = users_ds.merge(posts_ds)
users_posts_ds.head()

userId Reputation Views UpVotes DownVotes  postId  Score  ViewCount  \
0     -1          1     0    5007      1920    2175      0        NaN   
1     -1          1     0    5007      1920    8576      0        NaN   
2     -1          1     0    5007      1920    8578      0        NaN   
3     -1          1     0    5007      1920    8981      0        NaN   
4     -1          1     0    5007      1920    8982      0        NaN   

   CommentCount  
0             0  
1             0  
2             0  
3             0  
4             0

In [137]:
users_posts_ds.describe()

postId         Score      ViewCount  CommentCount
count  38962.000000  38962.000000   15390.000000  38962.000000
mean   22960.799651      4.083081    1196.225991      2.014630
std    13696.932471      6.561843    3742.952815      2.674018
min        1.000000    -19.000000      14.000000      0.000000
25%    11325.250000      1.000000     173.000000      0.000000
50%    22373.500000      2.000000     385.000000      1.000000
75%    33688.500000      5.000000     967.000000      3.000000
max    48325.000000    192.000000  175495.000000     45.000000

#### 16. How many missing values do you have in your merged dataframe? On which columns?

In [138]:
ds_null = users_posts_ds.isnull().sum()
ds_null

userId              0
Reputation          0
Views               0
UpVotes             0
DownVotes           0
postId              0
Score               0
ViewCount       23572
CommentCount        0
dtype: int64

In [139]:
ds_null [ds_null > 0]  

ViewCount    23572
dtype: int64

There are 23572 missing values from ViewCount DataFrame

#### 17. You will need to make something with missing values.  Will you clean or filling them? Explain. 
**Remember** to check the results of your code before passing to the next step

In [140]:
null_displ = users_posts_ds[(users_posts_ds['ViewCount'].isnull())]
null_displ.head()

userId Reputation Views UpVotes DownVotes  postId  Score  ViewCount  \
0     -1          1     0    5007      1920    2175      0        NaN   
1     -1          1     0    5007      1920    8576      0        NaN   
2     -1          1     0    5007      1920    8578      0        NaN   
3     -1          1     0    5007      1920    8981      0        NaN   
4     -1          1     0    5007      1920    8982      0        NaN   

   CommentCount  
0             0  
1             0  
2             0  
3             0  
4             0

In [141]:
null_displ

userId Reputation Views UpVotes DownVotes  postId  Score  ViewCount  \
0         -1          1     0    5007      1920    2175      0        NaN   
1         -1          1     0    5007      1920    8576      0        NaN   
2         -1          1     0    5007      1920    8578      0        NaN   
3         -1          1     0    5007      1920    8981      0        NaN   
4         -1          1     0    5007      1920    8982      0        NaN   
5         -1          1     0    5007      1920    9857      0        NaN   
6         -1          1     0    5007      1920    9858      0        NaN   
7         -1          1     0    5007      1920    9860      0        NaN   
8         -1          1     0    5007      1920   10130      0        NaN   
9         -1          1     0    5007      1920   10131      0        NaN   
10        -1          1     0    5007      1920   15331      0        NaN   
11        -1          1     0    5007      1920   15332      0        NaN   
12        -1          1     0    5007      1920   16296      0        NaN   
13        -1          1     0    5007      1920   16366      0        NaN   
14        -1          1     0    5007      1920   19423      0        NaN   
15        -1          1     0    5007      1920   19848      0        NaN   
16        -1          1     0    5007      1920   20259      0        NaN   
17        -1          1     0    5007      1920   20261      0        NaN   
18        -1          1     0    5007      1920   20262      0        NaN   
19        -1          1     0    5007      1920   20263      0        NaN   
20        -1          1     0    5007      1920   20264      0        NaN   
21        -1          1     0    5007      1920   20485      0        NaN   
22        -1          1     0    5007      1920   20487      0        NaN   
23        -1          1     0    5007      1920   20489      0        NaN   
24        -1          1     0    5007      1920   20490      0        NaN   
25        -1          1     0    5007      1920   21497      0        NaN   
26        -1          1     0    5007      1920   21795      0        NaN   
27        -1          1     0    5007      1920   21856      0        NaN   
28        -1          1     0    5007      1920   21860      0        NaN   
29        -1          1     0    5007      1920   21861      0        NaN   
...      ...        ...   ...     ...       ...     ...    ...        ...   
38903  25692        408    27       0         0    1213     47        NaN   
38905  26105        900   101      62         0   43222      1        NaN   
38912  26105        900   101      62         0   46599      2        NaN   
38914  26105        900   101      62         0   46766      2        NaN   
38916  26338       4800   515    1781        47   28238      0        NaN   
38917  26338       4800   515    1781        47   28328      0        NaN   
38918  26338       4800   515    1781        47   28978      0        NaN   
38919  26338       4800   515    1781        47   38854      0        NaN   
38920  26338       4800   515    1781        47   38877      0        NaN   
38921  26338       4800   515    1781        47   38879      0        NaN   
38922  26338       4800   515    1781        47   41312      0        NaN   
38923  27581       1159   112     186        12   20257      0        NaN   
38924  27581       1159   112     186        12   20258      0        NaN   
38925  27581       1159   112     186        12   28225      0        NaN   
38926  27581       1159   112     186        12   28265      0        NaN   
38927  27581       1159   112     186        12   28272      0        NaN   
38928  27581       1159   112     186        12   28302      0        NaN   
38929  27581       1159   112     186        12   28793      0        NaN   
38930  27581       1159   112     186        12   28901      0        NaN   
38931  27581       1159   112     186        12   28928      0        NaN   
38

In [142]:
ds_zeros = users_posts_ds[users_posts_ds == 0.0].count()
ds_zeros

userId              0
Reputation          0
Views            1150
UpVotes          6472
DownVotes       18036
postId              0
Score            4652
ViewCount           0
CommentCount    14983
dtype: int64

In [143]:
users_posts_ds['ViewCount'] = users_posts_ds['ViewCount'].fillna(0)
users_posts_ds

userId Reputation Views UpVotes DownVotes  postId  Score  ViewCount  \
0         -1          1     0    5007      1920    2175      0        0.0   
1         -1          1     0    5007      1920    8576      0        0.0   
2         -1          1     0    5007      1920    8578      0        0.0   
3         -1          1     0    5007      1920    8981      0        0.0   
4         -1          1     0    5007      1920    8982      0        0.0   
5         -1          1     0    5007      1920    9857      0        0.0   
6         -1          1     0    5007      1920    9858      0        0.0   
7         -1          1     0    5007      1920    9860      0        0.0   
8         -1          1     0    5007      1920   10130      0        0.0   
9         -1          1     0    5007      1920   10131      0        0.0   
10        -1          1     0    5007      1920   15331      0        0.0   
11        -1          1     0    5007      1920   15332      0        0.0   
12        -1          1     0    5007      1920   16296      0        0.0   
13        -1          1     0    5007      1920   16366      0        0.0   
14        -1          1     0    5007      1920   19423      0        0.0   
15        -1          1     0    5007      1920   19848      0        0.0   
16        -1          1     0    5007      1920   20259      0        0.0   
17        -1          1     0    5007      1920   20261      0        0.0   
18        -1          1     0    5007      1920   20262      0        0.0   
19        -1          1     0    5007      1920   20263      0        0.0   
20        -1          1     0    5007      1920   20264      0        0.0   
21        -1          1     0    5007      1920   20485      0        0.0   
22        -1          1     0    5007      1920   20487      0        0.0   
23        -1          1     0    5007      1920   20489      0        0.0   
24        -1          1     0    5007      1920   20490      0        0.0   
25        -1          1     0    5007      1920   21497      0        0.0   
26        -1          1     0    5007      1920   21795      0        0.0   
27        -1          1     0    5007      1920   21856      0        0.0   
28        -1          1     0    5007      1920   21860      0        0.0   
29        -1          1     0    5007      1920   21861      0        0.0   
...      ...        ...   ...     ...       ...     ...    ...        ...   
38932  27581       1159   112     186        12   33250      0        0.0   
38933  27581       1159   112     186        12   40831      0        0.0   
38934  27581       1159   112     186        12   40833      0        0.0   
38935  30893        116     2       0         0   19444      3       61.0   
38936  31017        160     6       6         0   46345      4     3221.0   
38937  31124        166     5       3         0   14544      6        0.0   
38938  31388         11     0       0         0   38872      1        0.0   
38939  32036       7227   991    2664       143   28258      0        0.0   
38940  35234        171     0       0         0   23445     14      840.0   
38941  35705         16     0       0         0   19577      3      409.0   
38942  35921        113     2       1         0   16397     -2        0.0   
38943  36602        121     0       2         0   27672      4     2244.0   
38944  37890        111     0       0         0   41240      1        0.0   
38945  40342        121     1       1         0   32863      4      144.0   
38946  42201        131     0       0         0   10683      3        0.0   
38947  43908        384    77      18         0   24072      2     2361.0   
38948  43908        384    77      18         0   24107      0      551.0   
38949  43908        384    77      18         0   25702      1      228.0   
38950  43908        384    77      18         0   25811     47     4463.0   
38951  43908        384    77      18         0   26326      3      820.0   
38

Since there are no zeros in the ViewCount collumn I decided to replace the NaN values for 0, since there could be a mistake in the registration of information. It is expectable that a post has 0 ViewCounts. 

#### 18. Adjust the data types in order to avoid future issues. Which ones should be changed? 

In [144]:
users_posts_ds.dtypes

userId           object
Reputation       object
Views            object
UpVotes          object
DownVotes        object
postId            int64
Score             int64
ViewCount       float64
CommentCount      int64
dtype: object

In [147]:
users_posts_ds = users_posts_ds.astype('int')

In [148]:
users_posts_ds

userId  Reputation  Views  UpVotes  DownVotes  postId  Score  \
0          -1           1      0     5007       1920    2175      0   
1          -1           1      0     5007       1920    8576      0   
2          -1           1      0     5007       1920    8578      0   
3          -1           1      0     5007       1920    8981      0   
4          -1           1      0     5007       1920    8982      0   
5          -1           1      0     5007       1920    9857      0   
6          -1           1      0     5007       1920    9858      0   
7          -1           1      0     5007       1920    9860      0   
8          -1           1      0     5007       1920   10130      0   
9          -1           1      0     5007       1920   10131      0   
10         -1           1      0     5007       1920   15331      0   
11         -1           1      0     5007       1920   15332      0   
12         -1           1      0     5007       1920   16296      0   
13         -1           1      0     5007       1920   16366      0   
14         -1           1      0     5007       1920   19423      0   
15         -1           1      0     5007       1920   19848      0   
16         -1           1      0     5007       1920   20259      0   
17         -1           1      0     5007       1920   20261      0   
18         -1           1      0     5007       1920   20262      0   
19         -1           1      0     5007       1920   20263      0   
20         -1           1      0     5007       1920   20264      0   
21         -1           1      0     5007       1920   20485      0   
22         -1           1      0     5007       1920   20487      0   
23         -1           1      0     5007       1920   20489      0   
24         -1           1      0     5007       1920   20490      0   
25         -1           1      0     5007       1920   21497      0   
26         -1           1      0     5007       1920   21795      0   
27         -1           1      0     5007       1920   21856      0   
28         -1           1      0     5007       1920   21860      0   
29         -1           1      0     5007       1920   21861      0   
...       ...         ...    ...      ...        ...     ...    ...   
38932   27581        1159    112      186         12   33250      0   
38933   27581        1159    112      186         12   40831      0   
38934   27581        1159    112      186         12   40833      0   
38935   30893         116      2        0          0   19444      3   
38936   31017         160      6        6          0   46345      4   
38937   31124         166      5        3          0   14544      6   
38938   31388          11      0        0          0   38872      1   
38939   32036        7227    991     2664        143   28258      0   
38940   35234         171      0        0          0   23445     14   
38941   35705          16      0        0          0   19577      3   
38942   35921         113      2        1          0   16397     -2   
38943   36602         121      0        2          0   27672      4   
38944   37890         111      0        0          0   41240      1   
38945   40342         121      1        1          0   32863      4   
38946   42201         131      0        0          0   10683      3   
38947   43908         384     77       18          0   24072      2   
38948   43908         384     77       18          0   24107      0   
38949   43908         384     77       18          0   25702      1   
38950   43908         384     77       18          0   25811     47   
38951   43908         384     77       18          0   26326      3   
38952   43908         384     77       18          0   26469      0   
38953   43908         384     77       18          0   26927      1   
38954   44618         128      1        0          0   21104      3   
38955   44915           1      0        0          0   34012      0   
38956   44995          11      

In [149]:
users_posts_ds.dtypes

userId          int64
Reputation      int64
Views           int64
UpVotes         int64
DownVotes       int64
postId          int64
Score           int64
ViewCount       int64
CommentCount    int64
dtype: object